### PR TITLE
feat(stackable-versioned): Add support for type changes of fields

### PR DESCRIPTION
**Pull request by Techassi**
Part of https://github.com/stackabletech/issues/issues/507

This PR adds support for type changes of struct fields. This is done by renaming the `renamed()` action to `changed()` and adding a new parameter `from_type` to it. Additionally, `from` will be renamed to `from_name`. 

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

### CODE REVIEW

**Comment by Techassi** [comment]
I would like to adjust some of the internal types and how the validation works. These changes are however not included in this PR, because it proved to be very difficult to implement the changes at the current moment. I will tackle this in future PRs.

It also keeps the PR small and focuses on the type change feature.

File: crates/stackable-versioned-macros/src/attrs/common/item.rs [hunk 1]
```diff
@@ -320,27 +311,33 @@ pub(crate) struct AddedAttributes {
...
+/// - `changed(since = "...", from_name = "...")`
+/// - `changed(since = "...", from_name = "..." from_type="...")`
 #[derive(Clone, Debug, FromMeta)]
-pub(crate) struct RenamedAttributes {
+pub(crate) struct ChangedAttributes {
     pub(crate) since: SpannedValue<Version>,
-    pub(crate) from: SpannedValue<String>,
+    pub(crate) from_name: Option<SpannedValue<String>>,
+
+    #[allow(dead_code)]
```
Comment (NickLarsenNZ) [inline crates/stackable-versioned-macros/src/attrs/common/item.rs:330]: :point_up:

File: crates/stackable-versioned-macros/src/codegen/common/item.rs [hunk 2]
```diff
@@ -125,116 +129,159 @@ where
...
                     ident: deprecated_ident.clone(),
-                    note: deprecated.note.to_string(),
+                    note: deprecated.note.as_deref().cloned(),
                 },
             );
 
-            for rename in common_attributes.renames.iter().rev() {
-                let from = format_ident!("{from}", from = *rename.from);
+            for change in common_attributes.changes.iter().rev() {
+                dbg!(&ty, &change.since);
```
Comment (NickLarsenNZ) [inline crates/stackable-versioned-macros/src/codegen/common/item.rs:156]: :point_up:

**Review by NickLarsenNZ — DISMISSED** [review]
LGTM

File: crates/stackable-versioned-macros/src/attrs/common/item.rs (hunk 1 above)
Comment (NickLarsenNZ) [reply crates/stackable-versioned-macros/src/attrs/common/item.rs:330]:
```suggestion
```

**Pull request merged**